actionpack/lib/action_dispatch/journey/router/utils.rb

```diff
@@ -35,7 +35,7 @@ class UriEncoder # :nodoc:
           US_ASCII = Encoding::US_ASCII
           UTF_8    = Encoding::UTF_8
           EMPTY    = (+"").force_encoding(US_ASCII).freeze
-          DEC2HEX  = (0..255).to_a.map { |i| ENCODE % i }.map { |s| s.force_encoding(US_ASCII) }
+          DEC2HEX  = (0..255).map { |i| (ENCODE % i).force_encoding(US_ASCII) }.freeze
```

Comment:
```suggestion
          DEC2HEX  = (0..255).map { |i| (ENCODE % i).force_encoding(US_ASCII) }
```

There is no need to freeze an Array, that is a constant. There is no advantage.